content/reference/template-functions/config-context.md

```diff
@@ -18,29 +18,47 @@ The config screen and associated options are described [here](/reference/v1beta1
 '{{repl ConfigOption "hostname" }}'
 ```
 
+`ConfigOption` returns the base64 **encoded** value of the `file` config option.
+
+```yaml
+'{{repl ConfigOption "ssl_key"}}'
+```
+
+To use files in a Secret, use `ConfigOption`:
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: tls-secret
+type: kubernetes.io/tls
+data:
+  tls.crt: '{{repl ConfigOption "tls_certificate_file" }}'
+  tls.key: '{{repl ConfigOption "tls_private_key_file" }}'
+```
+
+Learn more about [using TLS certs](/vendor/packaging/using-tls-certs) in KOTS.
+
 ## ConfigOptionData
 
 ```go
 func ConfigOptionData(fileName string) string
 ```
 
-Returns the base64 decoded value of a config option.
+`ConfigOptionData` returns the base64 **decoded** value of the `file` config option.
 
 ```yaml
 '{{repl ConfigOptionData "ssl_key"}}'
 ```
 
-This is often used to provide files as part of a secret or configmap, like this:
-
+To use files in a ConfigMap, use `ConfigOptionData`:
 ```yaml
 apiVersion: v1
-kind: Secret
+kind: ConfigMap
 metadata:
-  name: my-tls-secret
-type: kubernetes.io/tls
+  name: tls-config
 data:
-  tls.crt: '{{repl ConfigOptionData "tls_certificate_file" | Base64Encode }}'
-  tls.key: '{{repl ConfigOptionData "tls_private_key_file" | Base64Encode }}'
+  TLS_CRT: '{{repl ConfigOptionData "tls_certificate_file" }}'
+  TLS_KEY: '{{repl ConfigOptionData "tls_private_key_file" }}'
```

Comment:
@aj-jester what was the reasoning for switch from secret to config map? Often with non-Base64Encoded values for certs and keys, this will template out as 

```
  TLS_KEY: ---------------BEGIN CERTIFICATE-----------------
ABCDEFGHIJKLMNOPQRSTUVWXYZ1234568799
ABCDEFGHIJKLMNOPQRSTUVWXYZ1234568799
ABCDEFGHIJKLMNOPQRSTUVWXYZ1234568799
ABCDEFGHIJKLMNOPQRSTUVWXYZ1234568799
ABCDEFGHIJKLMNOPQRSTUVWXYZ1234568799
ABCDEFGHIJKLMNOPQRSTUVWXYZ1234568799
-----------------END CERTIFICATE-------------------

```

This breaks the yaml parser

Reply:
Arghh ignore me, just read the first section 🤕 , I get it now. I still think maybe a multiline file is a bad choice for ConfigMaps here, maybe we should just omit this example entirely? Or use a different kind of file example?

Reply:
Oh! sure, yah let me use an example without multi-lines now that I moved it to a different section.

Reply:
@dexhorthy Sure, let me add an example which does not involve multi-line, and instead I'll make a note on the multi lines with `| Base64Encode`.

Reply:
@dexhorthy I updated the `ConfigMap` to support multi-line files.

Reply:
@dexhorthy This now works as expected.

```
root@example-nginx-b79cd8797-kf7sn:/etc/nginx/ssl# cat tls.key

-----BEGIN PRIVATE KEY-----
MIIEvgIBADANBgkqhkiG9w0BAQEFAASCBKgwggSkAgEAAoIBAQC8iJ1GcqcdHZXD
NnFBHyxzqghF82vIoTXZ5wLAGYgptpMln1bf2Y3ZXeM6N5yElRaNUwKACWrYUb7d
LuS4x0KEAdweNLzoULqFkzZGbDIGTJ9eDDMzk/P1NbhlcCFfICfKOpi10GjBuSOQ
M/K4UpCvF+7ceyf6+5WsDEXZoHts1nmtI29KBxpHM/Luy7TX8Yj14WmQg3XGYekF
2diFZQV0ESvciPxpTHjFm8wA84D5QV7Gv8JUCuHQN5FtFWoXiiIpxRNMPWcDh9ee
A0Vy2AKruEeY+Ts3VXuTmStwht+wnO6tSstthBdkPdIxdaSNVLmqJTkx1AzU0WoU
1NQ1lwZJAgMBAAECggEAdi+BwBbBcbwJFO29A2DCYeN1+jvFFmkQs/9ftwKKW/PK
vGvtbrgXduqnNeiEoow/Leqah57frqlX55ms4bRvmwb7cKgIv+1Tb8a/4FeU6GQC
B1fIhQ27gbziUhx/nv0uVFR1RIxYDxC7HXujLS786fYr02UUw/4QbmDnLKywkqGw
QU+Q3IfUklMWK8nMITSLlMcFwbEv5hsuFbVVVfAGtbU+3B1/HJzbpdTYhpCrCa9L
suGKi7Cd7EPg8yqXYuA4LM4utFw3Phalykn//kvIDADFLHgIdECt5sVU4j3+Caa5
TELuv05wmezjrdV+xsfsUN0UcNlTuUJJTrTF5ZLmLQKBgQDsvmPP3bSM3FKM79Bf
67DDaEM/pj470xj7asX3Zvc0IKOt4cJAYQJYzaaYQI5IdQQTkQDY5mw3pRXL+STU
4gvPTQMgLWQmpsxfkiS10HgvP9naQPKLvGhRAXeNKYtsTSixDgmTW75GvQ3AtcIk
oRz46/Fd3LGkOJLzIIENKg3HqwKBgQDL3l4aHdj5HjzdSeA1GxusA+8K4pTTz7wh
ibu+ZxVirB4u1wmjb9KVDNknJKF0cmdFhqHyNn8uMQcn+CIX3RrnNeARjPy1vTZG
ofUd18ch2x3/ZCtrXTvHfZQ+te5baA97KZG4kc5KSRk24pdTAYF361O78JThS1Sm
l7ekglSl2wKBgHUtj021mMYXzyw/Ds5LZERz+MUqTadixvy6Tp4z8qt0dE9+bJ1c
5sJ5l7nISiSG9quQm/C/rFSme3FoMdjEwb+riJ6GmQmgiYvMVCtBshsA6Z1jhCmx
6Zget8PBCmdikX+BSTAMutNcg9AcpqlmrWZvshoZQDveXK3GKtoPNJ7dAoGBALKn
9B8++N2XOg9SpkDjuKRN7jXDyli0O1HIudZpz7vgybKSj8fCPvCPid1bpIG/n/C1
w0XXrCx2mqbUNExlXyYveGTpAKzjW6UaWQwGBGQO+YLN7Z7w3Wo9CqTKGk/QKYDt
NN5iKD3BQW9vstzy/NpyR7bzZgTpVw2G91AbRsLxAoGBAI7gmKkFDRs+HN+n9ETy
VEBGtRRvY9sRQ5AuGrV7O8pGXRLfX7NEVPn4HqAgEPL8A6RroAIO6fgZDLTrjBeH
PVk32NU7u/IP7rqxEKV//ktUQr1cFtnHzna2QyFIgC0crlyOIUF0MA/Tvpqyic8G
7hUjAfY85KBnxhc83zq88QhU
-----END PRIVATE KEY-----

root@example-nginx-b79cd8797-kf7sn:/etc/nginx/ssl# openssl rsa -noout -modulus -in tls.key
Modulus=BC889D4672A71D1D95C33671411F2C73AA0845F36BC8A135D9E702C0198829B693259F56DFD98DD95DE33A379C8495168D530280096AD851BEDD2EE4B8C7428401DC1E34BCE850BA859336466C32064C9F5E0C333393F3F535B86570215F2027CA3A98B5D068C1B9239033F2B85290AF17EEDC7B27FAFB95AC0C45D9A07B6CD679AD236F4A071A4733F2EECBB4D7F188F5E169908375C661E905D9D885650574112BDC88FC694C78C59BCC00F380F9415EC6BFC2540AE1D037916D156A178A2229C5134C3D670387D79E034572D802ABB84798F93B37557B93992B7086DFB09CEEAD4ACB6D8417643DD23175A48D54B9AA253931D40CD4D16A14D4D435970649
```